src/tools/rbd_mirror/group_replayer/BootstrapRequest.cc

```diff
@@ -1373,11 +367,14 @@ int BootstrapRequest<I>::create_replayers() {
       image_replayer->add_peer({local_pool_meta.mirror_uuid, remote_io_ctx,
                                 remote_pool_meta, m_remote_status_updater});
 
-      (*m_image_replayer_index)[{remote_pool_id, remote_image_id}] = image_replayer;
+//      m_image_replayers2->emplace(image_replayer, local_io_ctx);
+      //(*m_image_replayer_index)[{remote_pool_id, remote_image_id}] = image_replayer;
     }
-  } else if (m_local_mirror_group.state == cls::rbd::MIRROR_GROUP_STATE_ENABLED &&
-             m_local_mirror_group_primary) {
-    for (auto &[local_pool_id, global_image_id] : m_local_images) {
+  } else if ((*m_state_builder)->is_local_primary()) {
```

Comment:
Why try to create image replayers when the local is primary?

Reply:
That is required for the status updates which are required for the mirror pool status.

Reply:
Can you please add a code comment stating this?

Reply:
Done.